arraycontext/impl/jax/__init__.py

```diff
@@ -54,8 +54,8 @@ class and performs all array operations eagerly. See
     def __init__(self) -> None:
         super().__init__()
 
-        from jax.numpy import DeviceArray
-        self.array_types = (DeviceArray, )
+        from jax import Array as JAXArray
```

Comment:
Looking at the changelog (https://jax.readthedocs.io/en/latest/changelog.html), it seems like `jax.Array` was introduced in 0.3.18. 

Since this imports `jax.Array` unconditionally, maybe we should add a version constraint? i.e. `jax>0.3.18` or something?

Reply:
As far as I can see, the only installation-related place where `jax` is mentioned is in `.test-conda-env-py3.yml`. We could add the constraint there. On the other hand, if we used `jnp.ndarray`, would we need a constraint at all?

Reply:
Yeah, with `jnp.ndarray` it should be fine throughout :fingers-crossed: